compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -2272,9 +2270,7 @@ impl<D: Decoder> Decodable<D> for EncodedMetadata {
         let len = d.read_usize();
         let mmap = if len > 0 {
             let mut mmap = MmapMut::map_anon(len).unwrap();
-            for _ in 0..len {
-                (&mut mmap[..]).write_all(&[d.read_u8()]).unwrap();
-            }
+            mmap.copy_from_slice(d.read_raw_bytes(len));
             mmap.flush().unwrap();
```

Comment:
Huh. this shouldn't be necessary and it's probably bad for walltime because this triggers a writeback to disk and waits for it complete.
For cross-process visibility this shouldn't be necessary, assuming some implicit synchronization through the process ordering by cargo.

The only thing this would help with is OS crash-resilience/power loss. But we generally don't `sync_all` files, neither do we do the other durability dances, sometimes even bypass them. So we could skip that here too.
It might significantly reduce disk write pressure when those files are shortlived.

Reply:
#137032